evennia/contrib/rpg/rpsystem/rpsystem.py

```diff
@@ -1635,21 +1487,23 @@ def get_display_name(self, looker, **kwargs):
                 said object.
 
         Notes:
-            The RPCharacter version of this method colors its display to make
+            The RPCharacter version adds additional processing to sdescs to make
             characters stand out from other objects.
 
         """
-        idstr = "(#%s)" % self.id if self.access(looker, access_type="control") else ""
+        idstr = "(#%s)" % self.id if self.access(looker, access_type="control") and not kwargs.get("no_id",False) else ""
+        ref = kwargs.get("ref","~")
+    
         if looker == self:
-            sdesc = self.key
+            sdesc = self.process_recog(self.key,self)
```

Comment:
Why would you `recog` yourself?

Reply:
It's the processing code, not the actual recog handler - the logic there being, you recognize yourself, so it should be processed like other recognized character names. Although I can see an argument against differentiating between the two for the name line in `return_appearance` - it'd only take a small amount of rewriting, if you think that would be a better result.